crates/check/src/solution.rs

```diff
@@ -356,6 +411,92 @@ pub fn check_set_state_mutations(set: &SolutionSet) -> Result<(), InvalidSolutio
     Ok(())
 }
 
+/// Check the given solution set against the given predicates and
+/// and compute the post state mutations for this set.
+#[cfg_attr(feature = "tracing", tracing::instrument(skip_all))]
+pub async fn check_and_compute_solution_set<SA, SB>(
+    pre_state: &SA,
+    post_state: &SB,
+    solution_set: SolutionSet,
+    get_predicate: impl GetPredicate,
+    get_program: impl 'static + Clone + GetProgram + Send + Sync,
+    config: Arc<CheckPredicateConfig>,
+) -> Result<(Gas, SolutionSet), PredicatesError<SA::Error>>
+where
+    SA: Clone + StateRead + Send + Sync + 'static,
+    SB: Clone + StateRead<Error = SA::Error> + Send + Sync + 'static,
+    SA::Future: Send,
+    SB::Future: Send,
+    SA::Error: Send,
+{
+    // Check there are no existing mutations.
+    if solution_set
+        .solutions
+        .iter()
+        .any(|s| !s.state_mutations.is_empty())
+    {
+        return Err(PredicatesError::ExistingMutations);
+    }
```

Comment:
Ahh, so we make sure there are no mutations here anyway. I wonder if there's a case for allowing some existing ones when computing a new result, or if that might just cause confusion and it's safer to always deny it like this.

Reply:
Yeh I wasn't sure about this. It probably wouldn't make much sense because there would be no way to access them